controllers/storage/provision.go

```diff
@@ -158,17 +141,11 @@ func (m *MinIOReconciler) generateS3Secret(labels map[string]string) *corev1.Sec
 		Data: map[string][]byte{
 			s3Storage: dataJson,
 		},
-	}
+	},err
 }
 
-func (m *MinIOReconciler) generateAzureSecret(labels map[string]string) *corev1.Secret {
-	data := map[string]string{
-		"realm":       m.HarborCluster.Spec.Storage.Azure.Realm,
-		"accountname": m.HarborCluster.Spec.Storage.Azure.AccountName,
-		"accountkey":  m.HarborCluster.Spec.Storage.Azure.AccountKey,
-		"container":   m.HarborCluster.Spec.Storage.Azure.Container,
-	}
-	dataJson, _ := json.Marshal(&data)
+func (m *MinIOReconciler) generateAzureSecret(labels map[string]string) (*corev1.Secret, error) {
+	dataJson, err := json.Marshal(m.HarborCluster.Spec.Storage.Azure)
```

Comment:
Maybe dive a little bit of detail, why not return `nil, err` here and `&corev1.Secret{}, nil` at the end? Want to reduce code lines?